src/Factory.php

```diff
@@ -110,15 +111,12 @@ private function parseUrl($target)
             $parts['host'] = '127.0.0.1';
         }
 
-        $auth = null;
-        if (isset($parts['user'])) {
-            $auth = $parts['user'];
-        }
         if (isset($parts['pass'])) {
-            $auth .= ':' . $parts['pass'];
-        }
-        if ($auth !== null) {
-            $parts['auth'] = $auth;
+            // username:password@ (Redis doesn't support usernames)
+            $parts['auth'] = $parts['pass'];
+        } elseif (isset($parts['user'])) {
+            // password@
+            $parts['auth'] = $parts['user'];
```

Comment:
If we want to properly parse the URI, I suppose it makes sense to *only* support the "pass" component and ignore the "user" component entirely, so this block should probably be removed? :+1: 

We should also make sure to update the documentation and/or tests accordingly :+1: